### PR TITLE
[chore][testbed] - Further testbed enhancements

### PR DESCRIPTION
**Pull request by VihasMakwana**
PR to add memory limiter test cases.

This tests scenarios where collector is understress and we verify that we receive all the data, without drops/duplication.

### CODE REVIEW

**Comment by VihasMakwana** [comment]
@open-telemetry/collector-contrib-approvers can some take a look here?

**Comment by VihasMakwana** [comment]
ccing @atoulme as he reviewed my past tested PRs. Can you take a look?

**Comment by VihasMakwana** [comment]
@fatsheep9146 can you take a look here?

**Comment by VihasMakwana** [comment]
@open-telemetry/collector-contrib-approvers can someone take a look here?

**Review by atoulme — APPROVED** [review]

**Comment by atoulme** [comment]
This looks like a useful test to me. The test seems to test the right things.

**Review by djaglowski — APPROVED** [review]

File: testbed/tests/scenarios.go [hunk 1]
```diff
@@ -547,6 +547,95 @@ func ScenarioLong(
...
+	resultsSummary testbed.TestResultsSummary,
+	sleepTime int,
+	processors map[string]string,
+) {
+	resultDir, err := filepath.Abs(path.Join("results", t.Name()))
+	require.NoError(t, err)
+
+	agentProc := testbed.NewChildProcessCollector(testbed.WithEnvVar("GOMAXPROCS", "2"))
+
+	configStr := createConfigYaml(t, sender, receiver, resultDir, processors, nil)
```
Comment (songy23) [inline testbed/tests/scenarios.go:564]: This broke mainline https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/11727234196/job/32667863492

**Pull request merged**